src/sst/core/testingframework/test_engine_unittest.py

```diff
@@ -0,0 +1,534 @@
+            self.testresultdict[tm_tc] = SSTTestResultData()
+        return self.testresultdict[tm_tc]
+
+    def _get_test_module_test_case_name(self, test):
+        return "{0}.{1}".format(self._get_test_module_name(test),
+                                self._get_test_case_name(test))
+
+    def _get_test_case_name(self, test):
+        return strqual(test.__class__)
+
+    def _get_test_module_name(self, test):
+        return strclass(test.__class__)
+
+################################################################################
+
+class SSTTextTestResult(unittest.TextTestResult):
+    """ A superclass to support SST required testing """
+
+    def __init__(self, stream, descriptions, verbosity):
+        super(SSTTextTestResult, self).__init__(stream, descriptions, verbosity)
+        self.testresultdict = SSTTestResultDict()
+
+###
+
+    def startTest(self, test):
+        super(SSTTextTestResult, self).startTest(test)
+        #log_forced("DEBUG - startTest: Test = {0}\n".format(test))
+        self._start_time = time.time()
+        self._test_name = "undefined_testname"
+        _testName = getattr(test, 'testName', None)
+        if _testName is not None:
+            self._test_name = test.testName
+        self._testcase_name = test.get_testcase_name()
+        self._testsuite_name = test.get_testsuite_name()
+        timestamp = datetime.utcnow().strftime("%Y_%m%d_%H:%M:%S.%f utc")
+        self._junit_test_case = JUnitTestCase(self._test_name,
+                                              self._testcase_name,
+                                              timestamp=timestamp)
+
+    def stopTest(self, test):
+        super(SSTTextTestResult, self).stopTest(test)
+        #log_forced("DEBUG - stopTest: Test = {0}\n".format(test))
+        self._junit_test_case.junit_add_elapsed_sec(time.time() - self._start_time)
+        if not test_engine_globals.TESTENGINE_CONCURRENTMODE:
+            test_engine_globals.TESTRUN_JUNIT_TESTCASE_DICTLISTS['singlethread'].append(self._junit_test_case)
+        else:
+            test_engine_globals.TESTRUN_JUNIT_TESTCASE_DICTLISTS[self._testsuite_name].append(self._junit_test_case)
+
+###
+
+    def get_test_result_dict(self):
+        return self.testresultdict
+
+###
+
+    def addSuccess(self, test):
+        #super(SSTTextTestResult, self).addSuccess(test)
+        #log_forced("DEBUG - addSuccess: Test = {0}\n".format(test))
+        # Override the "ok" and make it a "PASS" instead
+        self.testresultdict.add_success(test)
+        if self.showAll:
+            self.stream.writeln("PASS")
+        elif self.dots:
+            self.stream.write('.')
+            self.stream.flush()
+
+    def addError(self, test, err):
+        super(SSTTextTestResult, self).addError(test, err)
+        self.testresultdict.add_error(test)
+        #log_forced("DEBUG - addError: Test = {0}, err = {1}\n".format(test, err))
+        _junit_test_case = getattr(self, '_junit_test_case', None)
```

Comment:
This is a trick that python unittest plays to check for extensions.  Due to the nature of how unittest works (complicated and convoluted), we cannot guarantee that the system will always be deriving from SSTTestCase under some rare conditions.  Because of this, the _junit_test_case object may not be an attribute of self and this would fail.